### ISSUE TITLE
mbed compile ..... -f doesn't flash device in Mbed CLI v1.9.0

### ISSUE THREAD

**Issue by MarceloSalazar** [ARMmbed/mbed-cli#851]
I believe this is a bug introduced in 1.9.0, as 1.8.3 works ok

When running 1.9.0:

```
$ mbed compile -t ARM -m target -f
...
Image: .\BUILD\target\ARM\mbed-os-example-blinky.bin
[mbed] ERROR: Unknown Error: local variable 'baudrate' referenced before assignment
```

An error is shown, and the target is not programmed anymore.

FYI @theotherjimmy @bridadan @screamerbg

**Comment by bridadan** [comment]
There was a regression in 1.9.0, but it was fixed in 1.9.1.

**Comment by MarceloSalazar** [comment]
indeed, the problem is gone in 1.9.1 - thanks!
